packages/hierarchy-builder/src/hierarchy-builder/internal/operators/grouping/PropertiesGrouping.ts

```diff
@@ -129,19 +146,12 @@ export async function createPropertyGroups(
       continue;
     }
 
-    const propertyClass = await getClass(metadata, byProperties.fullClassName);
-    const property = await propertyClass.getProperty(currentProperty.propertyName);
-    if (!property?.isPrimitive()) {
-      groupings.ungrouped.push(node);
-      continue;
-    }
-    const part = {
+    const formattedValue = await valueFormatter({
       type: property.primitiveType,
       extendedType: property.extendedTypeName,
       koqName: (await property.kindOfQuantity)?.fullName,
       value: currentProperty.propertyValue,
-    } as TypedPrimitiveValue;
-    const formattedValue = await valueFormatter(part);
+    } as TypedPrimitiveValue);
```

Comment:
is the `as TypedPrimitiveValue` necessary?

Reply:
![image](https://github.com/iTwin/presentation/assets/100586436/53d04311-8b1c-4b15-9517-929ea9c9879c)

**`property.primitiveType` is of this type:**

![image](https://github.com/iTwin/presentation/assets/100586436/1c5f48fd-d00a-452a-91d9-a11cf275c60c)

**Value formatter expect the `type` to be of this type:**
![image](https://github.com/iTwin/presentation/assets/100586436/42e15ddb-b719-461c-bf56-5079b721a489)

It doesn't accept 'Binary' and 'IGeometry' types.

Reply:
Well, that makes sense. We shouldn't cast just to make it think we're passing a valid type. I suggest handling `Binary` and `IGeometry` properties as if they were not primitive and place them into ungrouped list.